clang/lib/Driver/ToolChains/AVR.cpp

```diff
@@ -400,6 +400,14 @@ void AVRToolChain::AddClangSystemIncludeArgs(const ArgList &DriverArgs,
 void AVRToolChain::addClangTargetOptions(
     const llvm::opt::ArgList &DriverArgs, llvm::opt::ArgStringList &CC1Args,
     Action::OffloadKind DeviceOffloadKind) const {
+  // Reject C/C++ compilation for avr1 devices.
+  const Driver &D = getDriver();
+  std::string CPU = getCPUName(D, DriverArgs, getTriple());
```

Comment:
I think `const Driver &D = getDriver();` at line 404 is necessary, since I have to use  `D.Diag(..)`, it would not be better to call `getDriver()` twice.